python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -84,10 +85,23 @@ def asset2(asset1):
     check.opt_inst_param(_asset_selection_data, "_asset_selection_data", AssetSelectionData)
     source_assets_by_key = build_source_assets_by_key(source_assets)
 
-    deps, assets_defs_by_node_handle = build_deps(assets, source_assets_by_key.keys())
     partitioned_config = build_job_partitions_from_assets(assets, source_assets or [])
     resource_defs = check.opt_mapping_param(resource_defs, "resource_defs")
 
+    deps, assets_defs_by_node_handle = build_deps(assets, source_assets_by_key.keys())
+
+    try:
+        node_deps = {}
```

Comment:
Could be nice to put this inside a utility function called `has_cycles` or something.